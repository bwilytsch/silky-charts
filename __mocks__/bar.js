/* ignore file coverage */

export const data = [
  { name: 'foo', value: 19 },
  { name: 'bar', value: 70 },
  { name: 'baz', value: 93 },
  { name: 'quz', value: 80 },
];
export const dataWidthDates = [
  { name: '2018-01-31T05:00:00.000Z', value: 19 },
  { name: '2018-02-01T05:00:00.000Z', value: 70 },
  { name: '2018-02-02T05:00:00.000Z', value: 93 },
  { name: '2018-02-03T05:00:00.000Z', value: 80 },
];
