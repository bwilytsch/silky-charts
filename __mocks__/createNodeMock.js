export default element => {
  if (element.type === 'svg') {
    return {
      parentElement: {},
    };
  }
};
