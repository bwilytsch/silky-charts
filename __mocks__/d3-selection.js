/* ignore file coverage */

const selection = jest.genMockFromModule('d3-selection');

const attr = () => ({ attr });

const fn = () => ({
  call: jest.fn(),
  select: jest.fn(() => ({ attr })),
  selectAll: jest.fn(() => ({ attr })),
});

selection.select = fn;
selection.selectAll = fn;

module.exports = selection;
