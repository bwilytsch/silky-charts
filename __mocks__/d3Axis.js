/* ignore file coverage */

module.exports = jest.fn(() => ({
  ticks: jest.fn(() => ({
    tickFormat: jest.fn(),
  })),
}));
