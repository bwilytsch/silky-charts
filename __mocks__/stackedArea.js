/* ignore file coverage */

export const data = [
  { name: '2018-12-30T05:00:00.000Z', series: 'apples', value: 18 },
  { name: '2018-12-30T05:00:00.000Z', series: 'bananas', value: 16 },
  { name: '2018-12-30T05:00:00.000Z', series: 'cherries', value: 17 },
  { name: '2018-12-30T05:00:00.000Z', series: 'dates', value: 1 },
  { name: '2018-12-31T05:00:00.000Z', series: 'apples', value: 14 },
  { name: '2018-12-31T05:00:00.000Z', series: 'bananas', value: 14 },
  { name: '2018-12-31T05:00:00.000Z', series: 'cherries', value: 19 },
  { name: '2018-12-31T05:00:00.000Z', series: 'dates', value: 15 },
  { name: '2019-01-01T05:00:00.000Z', series: 'apples', value: 17 },
  { name: '2019-01-01T05:00:00.000Z', series: 'bananas', value: 13 },
  { name: '2019-01-01T05:00:00.000Z', series: 'cherries', value: 18 },
  { name: '2019-01-01T05:00:00.000Z', series: 'dates', value: 9 },
  { name: '2019-01-02T05:00:00.000Z', series: 'apples', value: 14 },
  { name: '2019-01-02T05:00:00.000Z', series: 'bananas', value: 13 },
  { name: '2019-01-02T05:00:00.000Z', series: 'cherries', value: 2 },
  { name: '2019-01-02T05:00:00.000Z', series: 'dates', value: 13 },
];
