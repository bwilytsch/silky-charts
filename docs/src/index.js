import React from 'react';
import { render } from 'react-dom';
import { Router, Redirect } from '@reach/router';
import Layout from './containers/Layout';
import Header from './containers/Header';
import Navigation from './containers/Navigation';
import Content from './containers/Content';
import Introduction from './components/Introduction';
import BarChart from './components/BarChart';
import pages from './pages.json';
import 'normalize.css';
import './styles.css';

const App = () => {
  const baseURL = '/silky-charts';
  const pathname = window.location.pathname;
  const currentPath = pathname.includes(baseURL)
    ? pathname.replace(baseURL, '')
    : '/';
  const isValidPathname =
    currentPath && pages.map(({ path }) => path).includes(currentPath);

  console.log(currentPath);

  const contentComponents = {
    Introduction,
    BarChart,
  };

  return (
    <Layout>
      <Header />
      <Navigation
        items={pages}
        baseURL={baseURL}
        initial={isValidPathname ? currentPath : '/'}
      />
      <Router primary={false}>
        {currentPath === '/' && (
          <Redirect from={currentPath} to={baseURL} noThrow />
        )}
        <Content path={baseURL}>
          {pages.map(({ component, path }, idx) => {
            const Component = contentComponents[component];
            return <Component key={idx} path={path} default={path === '/'} />;
          })}
        </Content>
      </Router>
    </Layout>
  );
};

render(<App />, document.getElementById('app'));
