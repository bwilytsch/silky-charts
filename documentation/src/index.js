import React from 'react';
import { render } from 'react-dom';
import { Router, Redirect } from '@reach/router';
import Layout from './containers/Layout';
import Header from './containers/Header';
import Navigation from './containers/Navigation';
import Content from './containers/Content';
import Introduction from './components/Introduction';
import BarChart from './components/BarChart';
import pages from './pages.json';
import 'normalize.css';
import './styles.css';

const App = () => {
  const pathname = window.location.pathname;
  const defaultPath = '/';
  const currentPath =
    pathname !== defaultPath ? pathname.replace(defaultPath, '') : pathname;
  const isInvalidPath = !pages
    .map(({ pathname }) => pathname)
    .includes(currentPath);

  const contentComponents = {
    Introduction,
    BarChart,
  };

  return (
    <Layout>
      <Header />
      <Navigation
        items={pages}
        initialPath={isInvalidPath ? defaultPath : currentPath}
      />
      <Content>
        <Router>
          {isInvalidPath && (
            <Redirect from={currentPath} to={defaultPath} noThrow />
          )}
          {pages.map(({ component, pathname }, idx) => {
            const Component = contentComponents[component];
            return (
              <Component
                key={idx}
                path={pathname}
                default={pathname === defaultPath}
              />
            );
          })}
        </Router>
      </Content>
    </Layout>
  );
};

render(<App />, document.getElementById('app'));
