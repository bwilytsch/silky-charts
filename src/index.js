export { default as Bar } from './Bar';
export { default as BarLine } from './BarLine';
export { default as StackedArea } from './StackedArea';

export { default as toSeries } from './utils/toSeries';
