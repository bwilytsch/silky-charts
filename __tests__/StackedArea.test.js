import React from 'react';
import { StackedArea } from '../src';
import { create } from 'react-test-renderer';
import { data, dataWidthDates } from '../__mocks__/stackedArea';
import createNodeMock from '../__mocks__/createNodeMock';
import 'jest-styled-components';

test('render correctly', () => {
  const tree = create(<StackedArea data={data} />, { createNodeMock }).toJSON();
  expect(tree).toMatchSnapshot();
});

test('render correctly with width', () => {
  const tree = create(<StackedArea data={data} width={400} />, {
    createNodeMock,
  }).toJSON();
  expect(tree).toMatchSnapshot();
});

test('render correctly with height', () => {
  const tree = create(<StackedArea data={data} height={400} />, {
    createNodeMock,
  }).toJSON();
  expect(tree).toMatchSnapshot();
});

test('render correctly with width and height', () => {
  const tree = create(<StackedArea data={data} width={400} height={300} />, {
    createNodeMock,
  }).toJSON();
  expect(tree).toMatchSnapshot();
});

test('render correctly with x axis label', () => {
  const tree = create(<StackedArea data={data} xAxisChartLabel="foo" />, {
    createNodeMock,
  }).toJSON();
  expect(tree).toMatchSnapshot();
});

test('render correctly with y axis label', () => {
  const tree = create(<StackedArea data={data} yAxisChartLabel="foo" />, {
    createNodeMock,
  }).toJSON();
  expect(tree).toMatchSnapshot();
});

test('render correctly with dates as name', () => {
  const tree = create(<StackedArea data={dataWidthDates} />, {
    createNodeMock,
  }).toJSON();
  expect(tree).toMatchSnapshot();
});

test('render correctly width horizontal grid', () => {
  const tree = create(<StackedArea data={data} grid />, {
    createNodeMock,
  }).toJSON();
  expect(tree).toMatchSnapshot();
});

test('render correctly width vertical grid', () => {
  const tree = create(<StackedArea data={data} grid horizontal />, {
    createNodeMock,
  }).toJSON();
  expect(tree).toMatchSnapshot();
});

test('render correctly X axis label rotation', () => {
  const tree = create(<StackedArea data={data} xAxisLabelRotation />, {
    createNodeMock,
  }).toJSON();
  expect(tree).toMatchSnapshot();
});
