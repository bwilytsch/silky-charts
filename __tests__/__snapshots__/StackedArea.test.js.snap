// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StackedArea 1`] = `
.c1 {
  fill: rgb(8,104,172);
  stroke: none;
  stroke-width: 3;
}

pointer-events: .c1:hover {
  fill: rgb(0,84,152);
}

.c2 {
  fill: rgb(67,162,202);
  stroke: none;
  stroke-width: 3;
}

pointer-events: .c2:hover {
  fill: rgb(47,142,182);
}

.c3 {
  fill: rgb(123,204,196);
  stroke: none;
  stroke-width: 3;
}

pointer-events: .c3:hover {
  fill: rgb(103,184,176);
}

.c4 {
  fill: rgb(186,228,188);
  stroke: none;
  stroke-width: 3;
}

pointer-events: .c4:hover {
  fill: rgb(166,208,168);
}

.c0 {
  height: 360px;
  width: 640px;
}

<svg
  baseProfile="full"
  class="silky-charts c0"
  xmlns="http://www.w3.org/2000/svg"
>
  <g
    class="silky-charts-container "
    transform="translate(50, 40)"
  >
    <g
      class="apples-layer"
    >
      <path
        class="line-path stacked-area c1"
        d="M0,191.61290322580646L180,209.03225806451613L360,195.96774193548387L540,209.03225806451613L540,270L360,270L180,270L0,270Z"
        stroke-width="0"
      />
    </g>
    <g
      class="bananas-layer"
    >
      <path
        class="line-path stacked-area c2"
        d="M0,121.93548387096774L180,148.06451612903226L360,139.3548387096774L540,152.4193548387097L540,209.03225806451613L360,195.96774193548387L180,209.03225806451613L0,191.61290322580646Z"
        stroke-width="0"
      />
    </g>
    <g
      class="cherries-layer"
    >
      <path
        class="line-path stacked-area c3"
        d="M0,47.90322580645159L180,65.32258064516128L360,60.96774193548387L540,143.70967741935485L540,152.4193548387097L360,139.3548387096774L180,148.06451612903226L0,121.93548387096774Z"
        stroke-width="0"
      />
    </g>
    <g
      class="dates-layer"
    >
      <path
        class="line-path stacked-area c4"
        d="M0,43.54838709677418L180,0L360,21.774193548387103L540,87.09677419354841L540,143.70967741935485L360,60.96774193548387L180,65.32258064516128L0,47.90322580645159Z"
        stroke-width="0"
      />
    </g>
    <g
      class="axis-x "
      transform="translate(0, 270)"
    />
    <g
      class="axis-y "
    />
  </g>
</svg>
`;

exports[`StackedArea 2`] = `
.c1 {
  fill: rgb(8,104,172);
  stroke: none;
  stroke-width: 3;
}

pointer-events: .c1:hover {
  fill: rgb(0,84,152);
}

.c2 {
  fill: rgb(67,162,202);
  stroke: none;
  stroke-width: 3;
}

pointer-events: .c2:hover {
  fill: rgb(47,142,182);
}

.c3 {
  fill: rgb(123,204,196);
  stroke: none;
  stroke-width: 3;
}

pointer-events: .c3:hover {
  fill: rgb(103,184,176);
}

.c4 {
  fill: rgb(186,228,188);
  stroke: none;
  stroke-width: 3;
}

pointer-events: .c4:hover {
  fill: rgb(166,208,168);
}

.c0 {
  height: 360px;
  width: 640px;
}

<svg
  baseProfile="full"
  class="silky-charts c0"
  xmlns="http://www.w3.org/2000/svg"
>
  <g
    class="silky-charts-container "
    transform="translate(50, 40)"
  >
    <g
      class="apples-layer"
    >
      <path
        class="line-path stacked-area c1"
        d="M0,191.61290322580646L180,209.03225806451613L360,195.96774193548387L540,209.03225806451613L540,270L360,270L180,270L0,270Z"
        stroke-width="0"
      />
    </g>
    <g
      class="bananas-layer"
    >
      <path
        class="line-path stacked-area c2"
        d="M0,121.93548387096774L180,148.06451612903226L360,139.3548387096774L540,152.4193548387097L540,209.03225806451613L360,195.96774193548387L180,209.03225806451613L0,191.61290322580646Z"
        stroke-width="0"
      />
    </g>
    <g
      class="cherries-layer"
    >
      <path
        class="line-path stacked-area c3"
        d="M0,47.90322580645159L180,65.32258064516128L360,60.96774193548387L540,143.70967741935485L540,152.4193548387097L360,139.3548387096774L180,148.06451612903226L0,121.93548387096774Z"
        stroke-width="0"
      />
    </g>
    <g
      class="dates-layer"
    >
      <path
        class="line-path stacked-area c4"
        d="M0,43.54838709677418L180,0L360,21.774193548387103L540,87.09677419354841L540,143.70967741935485L360,60.96774193548387L180,65.32258064516128L0,47.90322580645159Z"
        stroke-width="0"
      />
    </g>
    <g
      class="axis-x "
      transform="translate(0, 270)"
    />
    <g
      class="axis-y "
    />
  </g>
</svg>
`;

exports[`StackedArea width grid 1`] = `
.c1 path {
  stroke: transparent;
}

.c1 line {
  stroke: rgb(243,243,243);
}

.c2 {
  fill: rgb(8,104,172);
  stroke: none;
  stroke-width: 3;
}

pointer-events: .c2:hover {
  fill: rgb(0,84,152);
}

.c3 {
  fill: rgb(67,162,202);
  stroke: none;
  stroke-width: 3;
}

pointer-events: .c3:hover {
  fill: rgb(47,142,182);
}

.c4 {
  fill: rgb(123,204,196);
  stroke: none;
  stroke-width: 3;
}

pointer-events: .c4:hover {
  fill: rgb(103,184,176);
}

.c5 {
  fill: rgb(186,228,188);
  stroke: none;
  stroke-width: 3;
}

pointer-events: .c5:hover {
  fill: rgb(166,208,168);
}

.c0 {
  height: 360px;
  width: 640px;
}

<svg
  baseProfile="full"
  class="silky-charts c0"
  xmlns="http://www.w3.org/2000/svg"
>
  <g
    class="silky-charts-container "
    transform="translate(50, 40)"
  >
    <g
      class="grid c1"
    />
    <g
      class="apples-layer"
    >
      <path
        class="line-path stacked-area c2"
        d="M0,191.61290322580646L180,209.03225806451613L360,195.96774193548387L540,209.03225806451613L540,270L360,270L180,270L0,270Z"
        stroke-width="0"
      />
    </g>
    <g
      class="bananas-layer"
    >
      <path
        class="line-path stacked-area c3"
        d="M0,121.93548387096774L180,148.06451612903226L360,139.3548387096774L540,152.4193548387097L540,209.03225806451613L360,195.96774193548387L180,209.03225806451613L0,191.61290322580646Z"
        stroke-width="0"
      />
    </g>
    <g
      class="cherries-layer"
    >
      <path
        class="line-path stacked-area c4"
        d="M0,47.90322580645159L180,65.32258064516128L360,60.96774193548387L540,143.70967741935485L540,152.4193548387097L360,139.3548387096774L180,148.06451612903226L0,121.93548387096774Z"
        stroke-width="0"
      />
    </g>
    <g
      class="dates-layer"
    >
      <path
        class="line-path stacked-area c5"
        d="M0,43.54838709677418L180,0L360,21.774193548387103L540,87.09677419354841L540,143.70967741935485L360,60.96774193548387L180,65.32258064516128L0,47.90322580645159Z"
        stroke-width="0"
      />
    </g>
    <g
      class="axis-x "
      transform="translate(0, 270)"
    />
    <g
      class="axis-y "
    />
  </g>
</svg>
`;

exports[`StackedArea with Title, X label, Y label 1`] = `
.c2 {
  font-size: 0.9em;
  text-anchor: middle;
}

.c3 {
  font-size: 0.9em;
  text-anchor: middle;
  -webkit-transform: rotate(-90deg);
  -ms-transform: rotate(-90deg);
  transform: rotate(-90deg);
}

.c4 {
  fill: rgb(8,104,172);
  stroke: none;
  stroke-width: 3;
}

pointer-events: .c4:hover {
  fill: rgb(0,84,152);
}

.c5 {
  fill: rgb(67,162,202);
  stroke: none;
  stroke-width: 3;
}

pointer-events: .c5:hover {
  fill: rgb(47,142,182);
}

.c6 {
  fill: rgb(123,204,196);
  stroke: none;
  stroke-width: 3;
}

pointer-events: .c6:hover {
  fill: rgb(103,184,176);
}

.c7 {
  fill: rgb(186,228,188);
  stroke: none;
  stroke-width: 3;
}

pointer-events: .c7:hover {
  fill: rgb(166,208,168);
}

.c0 {
  height: 360px;
  width: 640px;
}

.c1 {
  font-size: 1.2em;
  text-anchor: middle;
}

<svg
  baseProfile="full"
  class="silky-charts c0"
  xmlns="http://www.w3.org/2000/svg"
>
  <g
    class="silky-charts-container "
    transform="translate(50, 40)"
  >
    <text
      class="chart-title c1"
      width="540"
      x="270"
      y="10"
    >
      Test title
    </text>
    <text
      class="x-axis-label c2"
      height="270"
      width="540"
      x="270"
      y="290"
    >
      Test X label
    </text>
    <text
      class="y-axis-label c3"
      height="270"
      width="540"
      x="-135"
      y="-16"
    >
      Test Y label
    </text>
    <g
      class="apples-layer"
    >
      <path
        class="line-path stacked-area c4"
        d="M0,191.61290322580646L180,209.03225806451613L360,195.96774193548387L540,209.03225806451613L540,270L360,270L180,270L0,270Z"
        stroke-width="0"
      />
    </g>
    <g
      class="bananas-layer"
    >
      <path
        class="line-path stacked-area c5"
        d="M0,121.93548387096774L180,148.06451612903226L360,139.3548387096774L540,152.4193548387097L540,209.03225806451613L360,195.96774193548387L180,209.03225806451613L0,191.61290322580646Z"
        stroke-width="0"
      />
    </g>
    <g
      class="cherries-layer"
    >
      <path
        class="line-path stacked-area c6"
        d="M0,47.90322580645159L180,65.32258064516128L360,60.96774193548387L540,143.70967741935485L540,152.4193548387097L360,139.3548387096774L180,148.06451612903226L0,121.93548387096774Z"
        stroke-width="0"
      />
    </g>
    <g
      class="dates-layer"
    >
      <path
        class="line-path stacked-area c7"
        d="M0,43.54838709677418L180,0L360,21.774193548387103L540,87.09677419354841L540,143.70967741935485L360,60.96774193548387L180,65.32258064516128L0,47.90322580645159Z"
        stroke-width="0"
      />
    </g>
    <g
      class="axis-x "
      transform="translate(0, 270)"
    />
    <g
      class="axis-y "
    />
  </g>
</svg>
`;

exports[`StackedArea with X label rotation 1`] = `
.c1 {
  fill: rgb(8,104,172);
  stroke: none;
  stroke-width: 3;
}

pointer-events: .c1:hover {
  fill: rgb(0,84,152);
}

.c2 {
  fill: rgb(67,162,202);
  stroke: none;
  stroke-width: 3;
}

pointer-events: .c2:hover {
  fill: rgb(47,142,182);
}

.c3 {
  fill: rgb(123,204,196);
  stroke: none;
  stroke-width: 3;
}

pointer-events: .c3:hover {
  fill: rgb(103,184,176);
}

.c4 {
  fill: rgb(186,228,188);
  stroke: none;
  stroke-width: 3;
}

pointer-events: .c4:hover {
  fill: rgb(166,208,168);
}

.c0 {
  height: 360px;
  width: 640px;
}

<svg
  baseProfile="full"
  class="silky-charts c0"
  xmlns="http://www.w3.org/2000/svg"
>
  <g
    class="silky-charts-container "
    transform="translate(50, 40)"
  >
    <g
      class="apples-layer"
    >
      <path
        class="line-path stacked-area c1"
        d="M0,191.61290322580646L180,209.03225806451613L360,195.96774193548387L540,209.03225806451613L540,270L360,270L180,270L0,270Z"
        stroke-width="0"
      />
    </g>
    <g
      class="bananas-layer"
    >
      <path
        class="line-path stacked-area c2"
        d="M0,121.93548387096774L180,148.06451612903226L360,139.3548387096774L540,152.4193548387097L540,209.03225806451613L360,195.96774193548387L180,209.03225806451613L0,191.61290322580646Z"
        stroke-width="0"
      />
    </g>
    <g
      class="cherries-layer"
    >
      <path
        class="line-path stacked-area c3"
        d="M0,47.90322580645159L180,65.32258064516128L360,60.96774193548387L540,143.70967741935485L540,152.4193548387097L360,139.3548387096774L180,148.06451612903226L0,121.93548387096774Z"
        stroke-width="0"
      />
    </g>
    <g
      class="dates-layer"
    >
      <path
        class="line-path stacked-area c4"
        d="M0,43.54838709677418L180,0L360,21.774193548387103L540,87.09677419354841L540,143.70967741935485L360,60.96774193548387L180,65.32258064516128L0,47.90322580645159Z"
        stroke-width="0"
      />
    </g>
    <g
      class="axis-x "
      transform="translate(0, 270)"
    />
    <g
      class="axis-y "
    />
  </g>
</svg>
`;

exports[`StackedArea with custom date format 1`] = `
.c1 {
  fill: rgb(8,104,172);
  stroke: none;
  stroke-width: 3;
}

pointer-events: .c1:hover {
  fill: rgb(0,84,152);
}

.c2 {
  fill: rgb(67,162,202);
  stroke: none;
  stroke-width: 3;
}

pointer-events: .c2:hover {
  fill: rgb(47,142,182);
}

.c3 {
  fill: rgb(123,204,196);
  stroke: none;
  stroke-width: 3;
}

pointer-events: .c3:hover {
  fill: rgb(103,184,176);
}

.c4 {
  fill: rgb(186,228,188);
  stroke: none;
  stroke-width: 3;
}

pointer-events: .c4:hover {
  fill: rgb(166,208,168);
}

.c0 {
  height: 360px;
  width: 640px;
}

<svg
  baseProfile="full"
  class="silky-charts c0"
  xmlns="http://www.w3.org/2000/svg"
>
  <g
    class="silky-charts-container "
    transform="translate(50, 40)"
  >
    <g
      class="apples-layer"
    >
      <path
        class="line-path stacked-area c1"
        d="M0,191.61290322580646L180,209.03225806451613L360,195.96774193548387L540,209.03225806451613L540,270L360,270L180,270L0,270Z"
        stroke-width="0"
      />
    </g>
    <g
      class="bananas-layer"
    >
      <path
        class="line-path stacked-area c2"
        d="M0,121.93548387096774L180,148.06451612903226L360,139.3548387096774L540,152.4193548387097L540,209.03225806451613L360,195.96774193548387L180,209.03225806451613L0,191.61290322580646Z"
        stroke-width="0"
      />
    </g>
    <g
      class="cherries-layer"
    >
      <path
        class="line-path stacked-area c3"
        d="M0,47.90322580645159L180,65.32258064516128L360,60.96774193548387L540,143.70967741935485L540,152.4193548387097L360,139.3548387096774L180,148.06451612903226L0,121.93548387096774Z"
        stroke-width="0"
      />
    </g>
    <g
      class="dates-layer"
    >
      <path
        class="line-path stacked-area c4"
        d="M0,43.54838709677418L180,0L360,21.774193548387103L540,87.09677419354841L540,143.70967741935485L360,60.96774193548387L180,65.32258064516128L0,47.90322580645159Z"
        stroke-width="0"
      />
    </g>
    <g
      class="axis-x "
      transform="translate(0, 270)"
    />
    <g
      class="axis-y "
    />
  </g>
</svg>
`;

exports[`StackedArea with height 1`] = `
.c1 {
  fill: rgb(8,104,172);
  stroke: none;
  stroke-width: 3;
}

pointer-events: .c1:hover {
  fill: rgb(0,84,152);
}

.c2 {
  fill: rgb(67,162,202);
  stroke: none;
  stroke-width: 3;
}

pointer-events: .c2:hover {
  fill: rgb(47,142,182);
}

.c3 {
  fill: rgb(123,204,196);
  stroke: none;
  stroke-width: 3;
}

pointer-events: .c3:hover {
  fill: rgb(103,184,176);
}

.c4 {
  fill: rgb(186,228,188);
  stroke: none;
  stroke-width: 3;
}

pointer-events: .c4:hover {
  fill: rgb(166,208,168);
}

.c0 {
  height: 100px;
  width: 177.77777777777777px;
}

<svg
  baseProfile="full"
  class="silky-charts c0"
  xmlns="http://www.w3.org/2000/svg"
>
  <g
    class="silky-charts-container "
    transform="translate(50, 40)"
  >
    <g
      class="apples-layer"
    >
      <path
        class="line-path stacked-area c1"
        d="M0,7.096774193548387L25.925925925925924,7.741935483870968L51.85185185185185,7.258064516129032L77.77777777777777,7.741935483870968L77.77777777777777,10L51.85185185185185,10L25.925925925925924,10L0,10Z"
        stroke-width="0"
      />
    </g>
    <g
      class="bananas-layer"
    >
      <path
        class="line-path stacked-area c2"
        d="M0,4.516129032258065L25.925925925925924,5.483870967741936L51.85185185185185,5.161290322580645L77.77777777777777,5.645161290322581L77.77777777777777,7.741935483870968L51.85185185185185,7.258064516129032L25.925925925925924,7.741935483870968L0,7.096774193548387Z"
        stroke-width="0"
      />
    </g>
    <g
      class="cherries-layer"
    >
      <path
        class="line-path stacked-area c3"
        d="M0,1.774193548387096L25.925925925925924,2.419354838709678L51.85185185185185,2.258064516129032L77.77777777777777,5.322580645161291L77.77777777777777,5.645161290322581L51.85185185185185,5.161290322580645L25.925925925925924,5.483870967741936L0,4.516129032258065Z"
        stroke-width="0"
      />
    </g>
    <g
      class="dates-layer"
    >
      <path
        class="line-path stacked-area c4"
        d="M0,1.612903225806452L25.925925925925924,0L51.85185185185185,0.806451612903226L77.77777777777777,3.225806451612904L77.77777777777777,5.322580645161291L51.85185185185185,2.258064516129032L25.925925925925924,2.419354838709678L0,1.774193548387096Z"
        stroke-width="0"
      />
    </g>
    <g
      class="axis-x "
      transform="translate(0, 10)"
    />
    <g
      class="axis-y "
    />
  </g>
</svg>
`;

exports[`StackedArea with source 1`] = `
.c2 {
  fill: rgb(8,104,172);
  stroke: none;
  stroke-width: 3;
}

pointer-events: .c2:hover {
  fill: rgb(0,84,152);
}

.c3 {
  fill: rgb(67,162,202);
  stroke: none;
  stroke-width: 3;
}

pointer-events: .c3:hover {
  fill: rgb(47,142,182);
}

.c4 {
  fill: rgb(123,204,196);
  stroke: none;
  stroke-width: 3;
}

pointer-events: .c4:hover {
  fill: rgb(103,184,176);
}

.c5 {
  fill: rgb(186,228,188);
  stroke: none;
  stroke-width: 3;
}

pointer-events: .c5:hover {
  fill: rgb(166,208,168);
}

.c1 {
  font-size: 0.7em;
  font-style: italic;
  text-anchor: end;
}

.c0 {
  height: 360px;
  width: 640px;
}

<svg
  baseProfile="full"
  class="silky-charts c0"
  xmlns="http://www.w3.org/2000/svg"
>
  <g
    class="silky-charts-container "
    transform="translate(50, 40)"
  >
    <text
      class="data-source c1"
      height="270"
      width="540"
      x="540"
      y="270"
    >
      Test source
    </text>
    <g
      class="apples-layer"
    >
      <path
        class="line-path stacked-area c2"
        d="M0,191.61290322580646L180,209.03225806451613L360,195.96774193548387L540,209.03225806451613L540,270L360,270L180,270L0,270Z"
        stroke-width="0"
      />
    </g>
    <g
      class="bananas-layer"
    >
      <path
        class="line-path stacked-area c3"
        d="M0,121.93548387096774L180,148.06451612903226L360,139.3548387096774L540,152.4193548387097L540,209.03225806451613L360,195.96774193548387L180,209.03225806451613L0,191.61290322580646Z"
        stroke-width="0"
      />
    </g>
    <g
      class="cherries-layer"
    >
      <path
        class="line-path stacked-area c4"
        d="M0,47.90322580645159L180,65.32258064516128L360,60.96774193548387L540,143.70967741935485L540,152.4193548387097L360,139.3548387096774L180,148.06451612903226L0,121.93548387096774Z"
        stroke-width="0"
      />
    </g>
    <g
      class="dates-layer"
    >
      <path
        class="line-path stacked-area c5"
        d="M0,43.54838709677418L180,0L360,21.774193548387103L540,87.09677419354841L540,143.70967741935485L360,60.96774193548387L180,65.32258064516128L0,47.90322580645159Z"
        stroke-width="0"
      />
    </g>
    <g
      class="axis-x "
      transform="translate(0, 270)"
    />
    <g
      class="axis-y "
    />
  </g>
</svg>
`;

exports[`StackedArea with source link 1`] = `
.c3 {
  fill: rgb(8,104,172);
  stroke: none;
  stroke-width: 3;
}

pointer-events: .c3:hover {
  fill: rgb(0,84,152);
}

.c4 {
  fill: rgb(67,162,202);
  stroke: none;
  stroke-width: 3;
}

pointer-events: .c4:hover {
  fill: rgb(47,142,182);
}

.c5 {
  fill: rgb(123,204,196);
  stroke: none;
  stroke-width: 3;
}

pointer-events: .c5:hover {
  fill: rgb(103,184,176);
}

.c6 {
  fill: rgb(186,228,188);
  stroke: none;
  stroke-width: 3;
}

pointer-events: .c6:hover {
  fill: rgb(166,208,168);
}

.c1 {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c2 {
  font-size: 0.7em;
  font-style: italic;
  text-anchor: end;
}

.c0 {
  height: 360px;
  width: 640px;
}

<svg
  baseProfile="full"
  class="silky-charts c0"
  xmlns="http://www.w3.org/2000/svg"
>
  <g
    class="silky-charts-container "
    transform="translate(50, 40)"
  >
    <a
      class="data-source-link c1"
      href="#"
      target="foo"
      title="stackedArea"
    >
      <text
        class="data-source c2"
        height="270"
        width="540"
        x="540"
        y="270"
      />
    </a>
    <g
      class="apples-layer"
    >
      <path
        class="line-path stacked-area c3"
        d="M0,191.61290322580646L180,209.03225806451613L360,195.96774193548387L540,209.03225806451613L540,270L360,270L180,270L0,270Z"
        stroke-width="0"
      />
    </g>
    <g
      class="bananas-layer"
    >
      <path
        class="line-path stacked-area c4"
        d="M0,121.93548387096774L180,148.06451612903226L360,139.3548387096774L540,152.4193548387097L540,209.03225806451613L360,195.96774193548387L180,209.03225806451613L0,191.61290322580646Z"
        stroke-width="0"
      />
    </g>
    <g
      class="cherries-layer"
    >
      <path
        class="line-path stacked-area c5"
        d="M0,47.90322580645159L180,65.32258064516128L360,60.96774193548387L540,143.70967741935485L540,152.4193548387097L360,139.3548387096774L180,148.06451612903226L0,121.93548387096774Z"
        stroke-width="0"
      />
    </g>
    <g
      class="dates-layer"
    >
      <path
        class="line-path stacked-area c6"
        d="M0,43.54838709677418L180,0L360,21.774193548387103L540,87.09677419354841L540,143.70967741935485L360,60.96774193548387L180,65.32258064516128L0,47.90322580645159Z"
        stroke-width="0"
      />
    </g>
    <g
      class="axis-x "
      transform="translate(0, 270)"
    />
    <g
      class="axis-y "
    />
  </g>
</svg>
`;

exports[`StackedArea with width 1`] = `
.c1 {
  fill: rgb(8,104,172);
  stroke: none;
  stroke-width: 3;
}

pointer-events: .c1:hover {
  fill: rgb(0,84,152);
}

.c2 {
  fill: rgb(67,162,202);
  stroke: none;
  stroke-width: 3;
}

pointer-events: .c2:hover {
  fill: rgb(47,142,182);
}

.c3 {
  fill: rgb(123,204,196);
  stroke: none;
  stroke-width: 3;
}

pointer-events: .c3:hover {
  fill: rgb(103,184,176);
}

.c4 {
  fill: rgb(186,228,188);
  stroke: none;
  stroke-width: 3;
}

pointer-events: .c4:hover {
  fill: rgb(166,208,168);
}

.c0 {
  height: 56.25px;
  width: 100px;
}

<svg
  baseProfile="full"
  class="silky-charts c0"
  xmlns="http://www.w3.org/2000/svg"
>
  <g
    class="silky-charts-container "
    transform="translate(50, 40)"
  >
    <g
      class="apples-layer"
    >
      <path
        class="line-path stacked-area c1"
        d="M0,-23.951612903225808L0,-26.129032258064516L0,-24.495967741935484L0,-26.129032258064516L0,-33.75L0,-33.75L0,-33.75L0,-33.75Z"
        stroke-width="0"
      />
    </g>
    <g
      class="bananas-layer"
    >
      <path
        class="line-path stacked-area c2"
        d="M0,-15.241935483870968L0,-18.508064516129032L0,-17.419354838709676L0,-19.05241935483871L0,-26.129032258064516L0,-24.495967741935484L0,-26.129032258064516L0,-23.951612903225808Z"
        stroke-width="0"
      />
    </g>
    <g
      class="cherries-layer"
    >
      <path
        class="line-path stacked-area c3"
        d="M0,-5.987903225806448L0,-8.16532258064516L0,-7.620967741935484L0,-17.963709677419356L0,-19.05241935483871L0,-17.419354838709676L0,-18.508064516129032L0,-15.241935483870968Z"
        stroke-width="0"
      />
    </g>
    <g
      class="dates-layer"
    >
      <path
        class="line-path stacked-area c4"
        d="M0,-5.443548387096772L0,0L0,-2.721774193548388L0,-10.887096774193552L0,-17.963709677419356L0,-7.620967741935484L0,-8.16532258064516L0,-5.987903225806448Z"
        stroke-width="0"
      />
    </g>
    <g
      class="axis-x "
      transform="translate(0, -33.75)"
    />
    <g
      class="axis-y "
    />
  </g>
</svg>
`;
